app/src/main/java/io/mochahub/powermeter/workoutsession/workoutsessiondialog/WorkoutSessionDialogViewModel.kt

```diff
@@ -53,10 +53,12 @@ class WorkoutSessionDialogViewModel(
                     workoutSessionUUID = workoutSessionEntity.id,
                     exerciseUUID = exercise.id
                 )
-
                 workoutEntities.add(workoutEntity)
                 workout.sets.forEach { workoutSet ->
                     if (workoutSet.weight != 0.0 && workoutSet.reps != 0) {
+                        if (workoutSet.weight>exercise.personalRecord) {
+                            db.exerciseDao().updateExercise(ExerciseEntity(exercise.id, exercise.name, exercise.muscleGroup, workoutSet.weight))
+                        }
```

Comment:
Think it would be a little cleaner if we made an extension function for the `ExerciseEntity` like:
```kotlin
fun ExerciseEntity.setPersonalRecord(newPersonalRecord:Double): ExerciseEntity{
    return this.copy(personalRecord = newPersonalRecord)
}
```
Then we can do:
```kotlin
db.exerciseDao().updateExercise(exercise.setPersonalRecord(workoutSet.weight))
```

Thoughts @brian-norman ?

Reply:
I agree, @mridulghanshala can you please update the PR?